pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -25,6 +35,151 @@ import (
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		objects := []runtime.Object{
+			&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "serving-cert", Namespace: "openshift-controller-manager"}},
+			&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "etcd-client", Namespace: "kube-system"}},
+			&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "client-ca", Namespace: "openshift-kube-apiserver"}},
+		}
+		kubeClient := fake.NewSimpleClientset(objects...)
+		cv := &configv1.ClusterVersion{}
+		configClientFake := configclientfake.NewSimpleClientset([]runtime.Object{cv}...)
+
+		indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+		proxyLister := configlistersv1.NewProxyLister(indexer)
+
+		operatorConfig := &operatorv1.OpenShiftControllerManager{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "cluster",
+				Generation: cv.Generation,
+			},
+			Spec: operatorv1.OpenShiftControllerManagerSpec{
+				OperatorSpec: operatorv1.OperatorSpec{},
+			},
+			Status: operatorv1.OpenShiftControllerManagerStatus{
+				OperatorStatus: operatorv1.OperatorStatus{},
+			},
+		}
+		kubeInformers := v1helpers.NewKubeInformersForNamespaces(kubeClient, "", util.TargetNamespace)
+		configMapsGetter := v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformers)
+		controllerManagerOperatorClient := operatorfake.NewSimpleClientset(operatorConfig)
+
+		configInformers := configinformers.NewSharedInformerFactory(configClientFake, 10*time.Minute)
+		configLister := configInformers.Config().V1().ClusterVersions().Lister()
+		operator := OpenShiftControllerManagerOperator{
+			kubeClient:           kubeClient,
+			configMapsGetter:     kubeClient.CoreV1(),
+			proxyLister:          proxyLister,
+			recorder:             events.NewInMemoryRecorder(""),
+			operatorConfigClient: controllerManagerOperatorClient.OperatorV1(),
+			configLister:         configLister,
```

Comment:
You can build the lister directly as in https://github.com/openshift/library-go/blob/d7e7beca5baea998564ed16bd3d7e7dffbc7c01f/pkg/operator/configobserver/featuregates/simple_featuregate_reader_test.go#L270-L279. E.g.
```go
indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
indexer.Add(cv)
configlistersv1.NewClusterVersionLister(indexer)
```

Reply:
please check